tls_main.c

```diff
@@ -567,42 +580,74 @@ static int do_tls_setsockopt_conf(struct sock *sk, char __user *optval,
 	return rc;
 }
 
-#define TLS_LUA 99
-#define lua_error(msg) pr_warn("[lua] %s - %s\n", __func__, msg);
-
-static int do_tls_loadlua(char __user *optval, unsigned int optlen)
+static int do_tls_loadlua(struct sock *sk, char __user *optval,
```

Comment:
This function is plagued by a bad design decision.
There's no need to keep the scripts in memory.
Lua will do exactly this, in a much more efficient form.
The user space program, after loading the script, should specify entry points to be called.
In our case, we should call an entry point (a Lua function) after we get the decrypted message.
We should have different `setsockopt`[1] options to deal with these scenarios, instead of funneling everything into this functions.

[1] I think we need only two. `TLS_LUA_LOADSCRIPT` and `TLS_LUA_ENTRYPOINT`.